### PR TITLE
Add templates for new pull requests and issues

### PR DESCRIPTION
**Pull request by tenpercent**
Motivation: provide structure for folks opening issues and pull requests.
Github blog on the feature: https://github.blog/developer-skills/github/issue-and-pull-request-templates/
TLDR:
> It’s hard to solve a problem when important details are missing. Now project maintainers can add templates for Issues and Pull Requests to projects, helping contributors share the right details…

> If you’re worried about the added clutter in the root directory of your project, we also added support for a .github/ folder. You can put `CONTRIBUTING.md`, `ISSUE_TEMPLATE.md`, and `PULL_REQUEST_TEMPLATE.md` files in .github/ and everything will work as expected.

The contents have been liberally compiled from https://github.com/devspace/awesome-github-templates?tab=readme-ov-file

### CODE REVIEW

**Review by spolifroni-amd — DISMISSED** [review]
Looks good from a docs perspective

**Comment by afagaj** [comment]
Does it make sense to unify [CONTRIBUTING.md](https://github.com/ROCm/composable_kernel/pull/1739/commits/545d20439e230346a610ea78e53d36eb7c19e9e8) with the [contributing-to-ck section](https://github.com/ROCm/composable_kernel?tab=readme-ov-file#contributing-to-ck) in the main README.md?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Please describe the motivation behind the pull request, whether it enables a new feature or fixes a bug. If there are associated pull requests or issues, please link them to the pull request.
+
+## Checklist
+
+Please put an `x` into the boxes that apply. You can also fill these out after creating the PR. If you're not sure, please don't hesitate to ask.
+
+- [ ] I have added tests relevant to the introduced functionality, and the unit tests are passing locally
+- [ ] I have added inline documentation which enables the maintainers with understanding the motivation
+- [ ] I have removed the stale documentation which is no longer relevant after this pull request
+- [ ] I have added release notes which provide the end users with a brief summary of the improvement from this pull request
```
Comment (afagaj) [inline .github/PULL_REQUEST_TEMPLATE.md]: Consider rewording from 
`[] I have added release notes which...` 
to something like 
`[] If this is a user visible and significant change, I have added release notes which..."`?

**Review by illsilin — APPROVED** [review]

**Comment by tenpercent** [comment]
> Does it make sense to unify [CONTRIBUTING.md](https://github.com/ROCm/composable_kernel/pull/1739/commits/545d20439e230346a610ea78e53d36eb7c19e9e8) with the [contributing-to-ck section](https://github.com/ROCm/composable_kernel?tab=readme-ov-file#contributing-to-ck) in the main README.md?

I think the `contributing-to-ck` section should be superseded by `contributing.md` file -- since the current content is about setting up git hooks to run the linter, which is not the only information a potential contributor needs. I've tried to create a good enough first level of references in `contributing.md`

**Review by aosewski — APPROVED** [review]

**Pull request merged**